src/clinical/service-worker-thread/tasks.ts

```diff
@@ -213,6 +216,35 @@ const mapEntityDocuments = (
   };
 };
 
+export function filterDonorIdDataFromSearch(donors: Donor[], query: ClinicalSearchQuery) {
+  const { donorIds, submitterDonorIds } = query;
+
+  const useFilteredDonors =
+    (donorIds && donorIds.length) || (submitterDonorIds && submitterDonorIds.length);
+
+  const filteredDonors = useFilteredDonors
+    ? donors.filter(donor => {
+        const { donorId, submitterId } = donor;
+        const stringId = `${donorId}`;
+        const donorMatch = donorIds?.filter(id => stringId.includes(id));
+        const submitterMatch = submitterId
+          ? submitterDonorIds?.filter(id => submitterId.includes(id))
+          : [];
+        return donorMatch.length > 0 || submitterMatch.length > 0;
```

Comment:
can likely do an early return if `donorMatch` finds what you're looking for, instead of both filter functions.
looks like it's just a check for a single match - can use `Array.some()` maybe so the whole array doesn't have to be iterated?